tests/functional/tier4/test_compile_linux_kernel_with_self_heal.py

```diff
@@ -0,0 +1,222 @@
+            ret, arequals = collect_mounts_arequal([self.mounts[0]],
+                                                   path=work_dir)
+            self.assertTrue(ret, 'Failed to get arequal')
+            g.log.info('Getting arequal before getting bricks offline '
+                       'is successful')
+
+            # Get arequal on bricks and compare with mount_point_total
+            self._check_arequal_on_bricks_with_a_specific_arequal(
+                arequals, dir_brick_list)
+
+        # Check arequals for "distributed-replicated"
+        if self.volume_type == "distributed-replicated":
+            # Get the subvolumes
+            subvols_dict = get_subvols(self.mnode, self.volname)
+            num_subvols = len(subvols_dict['volume_subvols'])
+
+            # Get arequals and compare
+            for i in range(0, num_subvols):
+                # Get arequal for first brick
+                brick_list = subvols_dict['volume_subvols'][i]
+                dir_brick_list = self._add_dir_path_to_brick_list(brick_list)
+                ret, arequals = collect_bricks_arequal([dir_brick_list[0]])
+                self.assertTrue(ret, 'Failed to get arequal on first brick')
+
+                # Get arequal for every brick and compare with first brick
+                self._check_arequal_on_bricks_with_a_specific_arequal(
+                    arequals, dir_brick_list)
+
+    def test_compile_linux_kernel_with_self_heal(self):
+        """
+        Test Case:
+        1. Create a volume of any type, start it and mount it.
+        2. Perform linux untar on the mount point and wait for it to complete.
+        3. Bring down brick processes of volume.
+        4. Compile linux kernel on monut point.
+        5. Bring offline bricks back online.
+        6. Wait for heal to complete.
+        7. Check arequal checksum on bricks and make sure there is
+           no data loss.
+
+        Note:
+        Run the below command before this test:
+        yum install elfutils-libelf-devel openssl-devel git fakeroot
+        ncurses-dev xz-utils libssl-dev bc flex libelf-dev bison
+
+        OR
+
+        dnf install make flex bison elfutils-libelf-devel openssl-devel
+        """
+        # Perform linux untar on the mount point and wait for it to complete
+        self._untar_linux_kernel_in_a_specific_dir()
+
+        # Bring down brick processes of volume
+        self._bring_bricks_offline()
+
+        # Compile linux kernel on monut point
+        cmd = ('cd {}/{}/linux-5.4.54/; make defconfig; make'.format(
+            self.mountpoint, 'test_self_heal'))
+        ret, _, _ = g.run(self.first_client, cmd)
+        self.assertFalse(ret, "Failed to compile linux kernel")
+        g.log.info("linux kernel compliation successful")
+
+        # Bring offline bricks back online
+        self._restart_volume_and_bring_all_offline_bricks_online()
+
+        # Wait for heal to complete
+        self._wait_for_heal_to_completed()
```

Comment:
This private method is used only once.  And its a wrapper of existing library. Please write the logic here itself.

Reply:
The intention of keeping it a private method is to reuse the same file for multiple similar cases. Another advantage to having it is that I don't what to disable pylint checks instead this helps us not hit them at all. Nacking this request as I feel it's invalid.

Reply:
This is the code present in the private function
        ret = monitor_heal_completion(self.mnode, self.volname,
                                      timeout_period=3600)
        self.assertTrue(ret, 'Heal has not yet completed')

I dont see any value add in writing a private function for this.

Reply:
```
ret = monitor_heal_completion(self.mnode, self.volname,
timeout_period=3600)
self.assertTrue(ret, 'Heal has not yet completed')
```
This three lines of code is pretty generic as we have use the same code for multiple cases where we would obly need to write one a single line instead of writing 3 x N lines where N is the number of tests which need it. This will help reduce code redundancy.